src/main/java/org/eclipse/tractusx/selfdescriptionfactory/config/ClearingHouseHealthIndicator.java

```diff
@@ -0,0 +1,48 @@
+/********************************************************************************
+ * Copyright (c) 2022 T-Systems International GmbH
+ * Copyright (c) 2022 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Apache License, Version 2.0 which is available at
+ * https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ********************************************************************************/
+
+package org.eclipse.tractusx.selfdescriptionfactory.config;
+
+import lombok.extern.slf4j.Slf4j;
+import org.eclipse.tractusx.selfdescriptionfactory.Utils;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.actuate.health.Health;
+import org.springframework.boot.actuate.health.HealthIndicator;
+import org.springframework.stereotype.Component;
+
+@Component("ClearingHouseHealthIndicator")
+@Slf4j
+public class ClearingHouseHealthIndicator implements HealthIndicator {
+
+    @Value("${app.usersDetails.clearingHouse.uri}")
+    private String url;
+
+    @Override
+    public Health health() {
+        if(Utils.checkUrlConnection(url)) {
+            return Health.up()
```

Comment:
Make sure that you don't try to connect to a url on the hot path of the health check but in parallel and return only the status of it

Reply:
@amoldashwant

Reply:
Removed this URL ping check as this is not really required to check the health of the SD factory.